src/state/routing/useRoutingAPITrade.ts

```diff
@@ -41,7 +42,7 @@ export function useRoutingAPITrade<TTradeType extends TradeType>(
   const queryArgs = useRoutingAPIArguments({
     tokenIn: currencyIn,
     tokenOut: currencyOut,
-    amount: amountSpecified,
+    amount: shouldGetTrade ? amountSpecified : undefined,
```

Comment:
we still want to interrupt polling when `shouldGetTrade` is false, so we've moved this into this hook.